flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -290,6 +292,12 @@
 	private final SqlValidatorImpl.ValidationErrorFunction validationErrorFunction =
 		new SqlValidatorImpl.ValidationErrorFunction();
 
+	// TypeCoercion instance used for implicit type coercion.
+	private TypeCoercion typeCoercion;
```

Comment:
Do we have to add this already even if we are not adding the support of type coercion now?

Reply:
Yes, this instance is default there, it does not bring in any overhead.